packages/ember-data/lib/serializers/json-serializer.js

```diff
@@ -124,6 +132,101 @@ export default Serializer.extend({
     return data;
   },
 
+  normalizeResponse: function(store, primaryTypeClass, payload, id, requestType) {
+    switch (requestType) {
+      case 'find':
+        return this.normalizeFindResponse(...arguments);
+      case 'findAll':
+        return this.normalizeFindAllResponse(...arguments);
+      case 'findBelongsTo':
+        return this.normalizeFindBelongsToResponse(...arguments);
+      case 'findHasMany':
+        return this.normalizeFindHasManyResponse(...arguments);
+      case 'findMany':
+        return this.normalizeFindManyResponse(...arguments);
+      case 'findQuery':
+        return this.normalizeFindQueryResponse(...arguments);
+      case 'createRecord':
+        return this.normalizeCreateRecordResponse(...arguments);
+      case 'deleteRecord':
+        return this.normalizeDeleteRecordResponse(...arguments);
+      case 'updateRecord':
+        return this.normalizeUpdateRecordResponse(...arguments);
+    }
```

Comment:
I have this vague memory that we wanted to change this to switch to make it easier for people to see what actual `requestType`s are getting passed. We just never got around to change it here. The discussion was originally around https://github.com/emberjs/data/blob/master/packages/ember-data/lib/adapters/build-url-mixin.js#L52-L75